core/block.js

```diff
@@ -1215,6 +1211,70 @@ Blockly.Block.prototype.interpolate_ = function(message, args, lastDummyAlign) {
 };
 
 /**
+ * Helper function to construct a FieldImage from a JSON arg object,
+ * dereferncing any string table references.
+ * @param {!Object} options A set of options (src, width, height, and alt).
+ * @returns {!Blockly.FieldImage} The new image.
+ */
+Blockly.Block.newFieldImageFromJson_ = function(options) {
+  var src = Blockly.utils.replaceMessageReferences(options['src']);
+  var width = options['width'];
+  if (goog.isString(width)) {
+    width = Number(Blockly.utils.replaceMessageReferences(width));
+  }
+  var height = options['height'];
+  if (goog.isString(height)) {
+    height = Number(Blockly.utils.replaceMessageReferences(height));
+  }
+  var alt = options['alt'];
+  if (goog.isString(alt)) {
+    alt = Blockly.utils.replaceMessageReferences(alt);
+  }
+  return new Blockly.FieldImage(src, width, height, alt);
+};
+
+/**
+ * Helper function to construct a FieldLabel from a JSON arg object,
+ * dereferncing any string table references.
+ * @param {!Object} options A set of options (text, and class).
+ * @returns {!Blockly.FieldImage} The new image.
+ */
+Blockly.Block.newFieldLabelFromJson_ = function(options) {
+  var text = Blockly.utils.replaceMessageReferences(options['text']);
+  return new Blockly.FieldLabel(text, options['class']);
+};
+
+/**
+ * Helper function to construct a FieldTextInput from a JSON arg object,
+ * dereferncing any string table references.
+ * @param {!Object} options A set of options (text, class, and spellcheck).
+ * @returns {!Blockly.FieldImage} The new image.
+ */
+Blockly.Block.newFieldTextInputFromJson_ = function(options) {
+  var text = Blockly.utils.replaceMessageReferences(options['text']);
+  var field = new Blockly.FieldTextInput(text, options['class']);
+  if (typeof options['spellcheck'] == 'boolean') {
+    field.setSpellcheck(options['spellcheck']);
+  }
+  return field;
+};
+
+/**
+ * Helper function to construct a FieldVariable from a JSON arg object,
+ * dereferncing any string table references.
+ * @param {!Object} options A set of options (variable).
+ * @returns {!Blockly.FieldImage} The new image.
+ */
+Blockly.Block.newFieldVariableFromJson_ = function(options) {
+  var varname = options['variable'];
+  if (goog.isString(varname)) {
```

Comment:
I think it'll always be a string or nothing, so you can just pass directly to Blockly.utils.replaceMessageReferences.